src/Nethermind/Nethermind.Merkleization/Merkleizer.cs

```diff
@@ -64,110 +64,11 @@ public void Feed(long value)
         Feed(_chunks[^1]);
     }
 
-    public void Feed(Span<byte> bytes, int? limit = null)
+    public void Feed<T>(Span<T> bytes, int? limit = null) where T : struct
     {
-        if (bytes.Length is not 0)
-        {
-            FeedAtLevel(MemoryMarshal.Cast<byte, UInt256>(bytes)[0], 0);
-        }
-        else
-        {
-            FeedAtLevel(UInt256.Zero, 0);
-        }
-        if (limit is not null)
-        {
-            Merkle.MixIn(ref _chunks[^1], limit.Value);
-        }
-    }
-    public void Feed(Span<short> bytes, int? limit = null)
-    {
-        if (bytes.Length is not 0)
-        {
-            FeedAtLevel(MemoryMarshal.Cast<short, UInt256>(bytes)[0], 0);
-        }
-        else
-        {
-            FeedAtLevel(UInt256.Zero, 0);
-        }
-        if (limit is not null)
-        {
-            Merkle.MixIn(ref _chunks[^1], limit.Value);
-        }
+        FeedAtLevel(bytes.Length is not 0 ? MemoryMarshal.Cast<T, UInt256>(bytes)[0] : UInt256.Zero, 0);
```

Comment:
btw don't we have to check if there is at least 32 bytes available?

Reply:
Yeah, need to do opposite - cast any type to byte array and then align it with 32. So non-generic and generic versions are buggy. Another problem is that with generic we need to explicitly mention generic parameter when we call Feed which makes it not very fun